### PR TITLE
Update topaz API and directory configuration

### PR DESCRIPTION
**Pull request by carabasdaniel**
Closes https://github.com/aserto-dev/topaz/issues/92

### CODE REVIEW

**Comment by gertd** [comment]
We should hold this change until we rework the service hosting to solve the race condition starting topaz as that most likely will result in a configuration change, as topaz needs to be able to configure all network endpoints and services associated with the endpoints.

File: pkg/app/authorizer.go [hunk 1]
```diff
@@ -2,59 +2,318 @@ package app
...
+		} else {
+			serv.registeredServices = append(serv.registeredServices, key)
+			serv.API = config
+		}
+		portMap[config.GRPC.ListenAddress] = serv
+	}
+	return portMap
+}
+
+func contains(slice []string, item string) bool {
```
Comment (ronenh) [inline pkg/app/authorizer.go:272]: :art:
```go
func contains[T comparable](slice []T, item T) bool {
    for i := range slice {
		if slice[i] == item {
			return true
		}
	}
	return false
}
```

File: pkg/builder/service_factory.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	}
+
+	err = registrations(context.Background(), runtimeMux, grpcEndpoint, opts)
+	if err != nil {
+		return Gateway{}, err
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/", runtimeMux)
+	mux.Handle("/api/", fieldsMaskHandler(runtimeMux))
```
Comment (ronenh) [inline pkg/builder/service_factory.go:91]: Maybe not an immediate blocker, but we should provide a way to expose OpenAPI specs services that have a REST API.

Comment (carabasdaniel) [reply pkg/builder/service_factory.go:91]: For now the service's gateway mux is public after the service creation and I'm currently attaching the openapi handlers in the authorizer initialization, but I think we can improve on this.

File: go.mod [hunk 3]
```diff
@@ -14,21 +16,21 @@ require (
 	github.com/aserto-dev/errors v0.0.5
 	github.com/aserto-dev/go-aserto v0.20.3
 	github.com/aserto-dev/go-authorizer v0.20.2
-	github.com/aserto-dev/go-directory v0.21.0
+	github.com/aserto-dev/go-directory v0.20.8-0.20230619123658-71cf8ee09ffc
```
Comment (ronenh) [inline go.mod:19]: We may need to switch back to the proto version without the REST binding if we want to release this soon.
In a conversation with Gert and Omri we decided to introduce a v3 of the directory API with a few breaking changes to make certain optional inputs required and base the REST mapping on that v3 API in order to expose a more idiomatic API with GET, PUT, and DELETE requests on objects and relations.

File: docs/config.md [hunk 4]
```diff
@@ -101,14 +129,13 @@ The remote address can also be configured to a service that implements the proto
 
 Example (using the hosted Aserto directory):
 ```
-directory_service:
-  remote:
-    address: "directory.prod.aserto.com:8443"
-    api_key: <Your Aserto Directory Access Key>
-    tenant_id: <Your Aserto Tenant ID>
+remote:
```
Comment (ronenh) [inline docs/config.md:132]: Shouldn't this be `remote_directory:`?

**Review by gertd — APPROVED** [review]

**Pull request merged**